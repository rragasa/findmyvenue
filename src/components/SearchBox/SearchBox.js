import React, { Component } from 'react';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import InputBase from '@material-ui/core/InputBase';
import Box from '@material-ui/core/Box';
import IconButton from '@material-ui/core/IconButton';
import SearchIcon from '@material-ui/icons/Search';
class SearchBox extends Component {
  constructor(props) {
    super(props);
    this.state = {location: ''};
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }

  handleSubmit(event) {
    const { onSubmit } = this.props;
    event.preventDefault();
    onSubmit(this.state.location);
  }

  handleChange(event) {
    this.setState({ location: event.target.value });
  }

  render() {
    const {location} = this.state;
    return (
      <div className="component-container">
        <header className="App-header">
          <Typography
            variant="h4"
            component="h4" gutterBottom>Find my venue</Typography>
        </header>
        <form onSubmit={this.handleSubmit}>
        <Paper>
          <Box display="flex">
            <InputBase
              className="space-left"
              placeholder="Explore venues..."
              inputProps={{ 'aria-label': 'Explore venues' }}
              value={location}
              fullWidth
              margin="dense"
              onChange={e => this.handleChange(e)}
            />
            <IconButton onClick={this.handleSubmit} aria-label="search">
              <SearchIcon />
            </IconButton>
          </Box>
        </Paper>
      </form>
      </div>
    );
  }
}

export default SearchBox;