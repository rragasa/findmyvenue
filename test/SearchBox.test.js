import React from 'react';
import { shallow } from 'enzyme';
import SearchBox from '../src/components/SearchBox';

describe('SearchBox Component', () => {
  const props = {
    onSubmit: jest.fn(),
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<SearchBox {...props} />);
  });

  it('should render the SearchBox component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the search button', () => {
    wrapper.find('[aria-label="search"]')
      .simulate('click', { preventDefault: () => {} });
    expect(props.onSubmit).toHaveBeenCalled();
  });
});
