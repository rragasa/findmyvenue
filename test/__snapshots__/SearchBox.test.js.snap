// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render the app component 1`] = `ShallowWrapper {}`;

exports[`SearchBox Component should render the SearchBox component 1`] = `ShallowWrapper {}`;
